singularity-eos/eos/eos_spiner_rho_sie.hpp

```diff
@@ -60,6 +60,53 @@ namespace singularity {
 
 using namespace eos_base;
 
+
+static PORTABLE_FORCEINLINE_FUNCTION Real toLog_(const Real x, const Real offset) {
+   // return std::log10(std::abs(std::max(x,-offset) + offset)+robust::EPS());
+   return FastMath::log10(std::abs(std::max(x, -offset) + offset) + robust::EPS());
+}
+static PORTABLE_FORCEINLINE_FUNCTION Real fromLog_(const Real lx, const Real offset) {
+   return FastMath::pow10(lx) - offset;
+}
+
+
+struct Transform_Data {
+  static constexpr int NGRIDS = 3;
+  using Grid_t = Spiner::PiecewiseGrid1D<Real, NGRIDS>;
+  using DataBox = Spiner::DataBox<Real, Grid_t>;
+
+  Real lRhoOffset;
+  DataBox sieCold;
+
+  PORTABLE_INLINE_FUNCTION
+  Transform_Data()
+    : lRhoOffset(0.0), sieCold() 
+  {}
+
+  PORTABLE_INLINE_FUNCTION
+  Transform_Data(Real lRhoOffset_, const DataBox &sieCold_) : lRhoOffset(lRhoOffset_), sieCold(sieCold_) {}
+};
+
+template <typename Data = void>
+struct NullTransform {
+  PORTABLE_INLINE_FUNCTION
+  NullTransform(const Data& = Data{}) {}
```

Comment:
A more extensible (and readable in my opinion) version would be to use a variadic template to communicate that this class doesn't need any arguments. This would also allow us to change (if we wanted to) how the transformers are instantiated without having to modify this class.
```suggestion
template <typename DataT = void>
struct NullTransform {
  PORTABLE_INLINE_FUNCTION
  template<typename... Args>
  NullTransform(Args&&...) {}
```

Reply:
I would also add 
```C++
PORTABLE_INLINE_FUNCTION
NullTransform() = default;
```
to make this one default constructible. (Though maybe that will conflict with the variadic constructor, we'd have to see.)